File: app/src/main/java/com/example/pricerunner/RecyclerAdapter.java
package com.example.pricerunner;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.ViewHolder> {

    private ArrayList<Information> productList;
    private Context context;

    public RecyclerAdapter(ArrayList<Information>productList, Context context) {
        this.productList = productList;
        this.context = context;
    }

    @NonNull
    @Override
    public RecyclerAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_design,parent,false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        holder.priceTextView.setText(productList.get(position).getPrice());
        holder.titleTextView.setText(productList.get(position).getTitle());
        holder.sellerTextView.setText(productList.get(position).getSeller());
        Picasso.get().load(productList.get(position).getImageURL()).resize(100,80).into(holder.productImageView);

    }

    @Override
    public int getItemCount() {
        return productList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private ImageView productImageView;
        private TextView titleTextView, priceTextView, sellerTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            productImageView = (ImageView) itemView.findViewById(R.id.imageViewProduct);
            titleTextView = (TextView) itemView.findViewById(R.id.tvProductName);
            priceTextView = (TextView) itemView.findViewById(R.id.tvProductPrice);
            sellerTextView = (TextView) itemView.findViewById(R.id.tvSellerName);
        }
    }
}

File: app/src/main/java/com/example/pricerunner/RecyclerAdapter.kt
package com.example.pricerunner

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso

class RecyclerAdapter(
    private val productList: List<Information>
) : RecyclerView.Adapter<RecyclerAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.card_design, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val product = productList[position]
        holder.titleTextView.text = product.title
        holder.priceTextView.text = product.price
        holder.sellerTextView.text = product.seller
        
        Picasso.get()
            .load(product.image)
            .resize(100, 80)
            .into(holder.productImageView)
    }

    override fun getItemCount(): Int = productList.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val productImageView: ImageView = itemView.findViewById(R.id.imageViewProduct)
        val titleTextView: TextView = itemView.findViewById(R.id.tvProductName)
        val priceTextView: TextView = itemView.findViewById(R.id.tvProductPrice)
        val sellerTextView: TextView = itemView.findViewById(R.id.tvSellerName)
    }
}
